src/lib/core/overlay/position/connected-position-strategy.spec.ts

```diff
@@ -29,7 +41,11 @@ describe('ConnectedPositionStrategy', () => {
     let overlayContainerElement: HTMLElement;
     let strategy: ConnectedPositionStrategy;
     let fakeElementRef: ElementRef;
-    let fakeViewportRuler: FakeViewportRuler;
+    // TODO: The `any` type is used here because the FakeViewportRuler can't extend or implement
+    // the real ViewportRuler class because Typescript complains when using a class as an interface.
+    // Also we don't want to extend the ViewportRuler because we'd need to call the constructor
+    // in the superclass. In short: We should use Dependency Injection to fake the ViewportRuler.
+    let fakeViewportRuler: any;
```

Comment:
TypeScript is fine with doing `implements RealClass`; it's one of the language features. The problem is the new `private` member, which causes any other implementation to automatically fail to match. You should be able to resolve this by marking the private member as optional ([example](http://www.typescriptlang.org/play/#src=class%20ViewportRuler%20%7B%0A%20%20%20%20private%20_cache%3F%3A%20number%3B%0A%0A%20%20%20%20measure()%3A%20number%20%7B%0A%20%20%20%20%20%20%20%20return%207%3B%0A%20%20%20%20%7D%0A%7D%0A%0Aclass%20FakeViewportRuler%20implements%20ViewportRuler%20%7B%0A%20%20%20%20measure()%20%7B%0A%20%20%20%20%20%20%20%20return%209%3B%0A%20%20%20%20%7D%0A%7D#src=class%20ViewportRuler%20%7B%0A%20%20%20%20private%20_cache%3F%3A%20number%3B%0A%0A%20%20%20%20_remember%3F()%20%7B%0A%20%20%20%20%20%20%20%20this._cache%20%3D%207%3B%0A%20%20%20%20%7D%0A%0A%20%20%20%20measure()%3A%20number%20%7B%0A%20%20%20%20%20%20%20%20return%207%3B%0A%20%20%20%20%7D%0A%7D%0A%0Aclass%20FakeViewportRuler%20implements%20ViewportRuler%20%7B%0A%20%20%20%20measure()%20%7B%0A%20%20%20%20%20%20%20%20return%209%3B%0A%20%20%20%20%7D%0A%7D)).

(and the internal method as well)